pubnub-gson/src/main/java/com/pubnub/internal/endpoints/DelegatingEndpoint.java

```diff
@@ -0,0 +1,75 @@
+package com.pubnub.internal.endpoints;
+
+import com.pubnub.api.PubNubException;
+import com.pubnub.api.endpoints.Endpoint;
+import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
+import com.pubnub.api.enums.PNOperationType;
+import com.pubnub.api.v2.callbacks.Result;
+import com.pubnub.internal.PubNubCore;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Consumer;
+
+public abstract class DelegatingEndpoint<T> implements Endpoint<T> {
+    protected final PubNubCore pubnub;
+    private volatile ExtendedRemoteAction<T> remoteAction = null;
+
+    public DelegatingEndpoint(PubNubCore pubnub) {
+        this.pubnub = pubnub;
+    }
+
+    protected abstract ExtendedRemoteAction<T> createAction();
+
+    public T sync() throws PubNubException {
+        validateParams();
+        return getRemoteAction().sync();
+    }
+
+    protected void validateParams() throws PubNubException {
```

Comment:
There are many classes that extends DelegatingEndpoint but only few of them like SignalImpl.java override validateParams() method. I am wondering how it is possible?
I can see that there is validation in SignalImpl.java:

```
    @Override
    protected void validateParams() throws PubNubException {
        if (message == null) {
            throw new PubNubException(PubNubErrorBuilder.PNERROBJ_MESSAGE_MISSING);
        }
        if (channel == null || channel.isEmpty()) {
            throw new PubNubException(PubNubErrorBuilder.PNERROBJ_CHANNEL_MISSING);
        }
    }
```

and then there is validation in SignalEndpoint.kt that kind of check channel being not blank again
```
    override fun validateParams() {
        super.validateParams()
        if (channel.isBlank()) throw PubNubException(PubNubError.CHANNEL_MISSING)
    }

```

Reply:
added back validations for java sdk